AppSyncRealTimeClient/ConnectionProvider/AppsyncRealtimeConnection/RealtimeConnectionProviderResponse.swift

```diff
@@ -95,4 +95,25 @@ extension RealtimeConnectionProviderResponse {
         return false
     }
 
+    func isLimitExceededError() -> Bool {
+        // It is expected to contain payload with corresponding error information
+        guard let payload = payload else {
+            return false
+        }
+
+        // The observed response from the service
+        // { "type":"error",
+        //  "payload": {
+        //      "errors": {
+        //          "errorType":"LimitExceededError",
+        //          "message":"Rate limit exceeded" }}}
+        if let errors = payload["errors"],
+           case let .object(errorsObject) = errors,
+           let errorType = errorsObject["errorType"],
+           errorType == "LimitExceededError" {
```

Comment:
nit: it might be worth to extract all these error types (`LimitExceededError`, `MaxSubscriptionsReachedError`, `MaxSubscriptionsReachedException`) in an enum

Reply:
I can do this in the next PR -> https://github.com/aws-amplify/aws-appsync-realtime-client-ios/pull/69/files it is also error handling related